support-workers/src/test/scala/com/gu/acquisitions/BigQuerySpec.scala

```diff
@@ -0,0 +1,80 @@
+package com.gu.acquisitions
+
+import com.google.auth.oauth2.ServiceAccountCredentials
+import com.google.cloud.bigquery.{BigQueryOptions, QueryJobConfiguration}
+import com.gu.config.Configuration
+import com.gu.i18n.{Country, Currency}
+import com.gu.salesforce.Salesforce.{Authentication, DeliveryContact, NewContact, SalesforceContactResponse}
+import com.gu.support.acquisitions.AcquisitionType.Purchase
+import com.gu.support.acquisitions.{AbTest, AcquisitionDataRow, AcquisitionProduct, BigQueryService, QueryParameter}
+import com.gu.support.workers.{Monthly, PayPal}
+import com.gu.support.zuora.api.ReaderType.Direct
+import com.gu.test.tags.annotations.IntegrationTest
+import com.typesafe.scalalogging.LazyLogging
+import org.joda.time.{DateTime, DateTimeZone}
+import org.joda.time.format.ISODateTimeFormat
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.collection.JavaConverters._
+
+@IntegrationTest
+class BigQuerySpec extends AsyncFlatSpec with Matchers with LazyLogging {
+  lazy val config = Configuration.load().bigQueryConfigProvider.get()
+  lazy val bigQuery =
+    BigQueryOptions
+      .newBuilder()
+      .setProjectId(config.projectId)
+      .setCredentials(ServiceAccountCredentials.fromPkcs8(
+        config.clientId,
+        config.clientEmail,
+        config.privateKey,
+        config.privateKeyId,
+        Nil.asJavaCollection
+      ))
+      .build().getService
+
+  "BigQuery" should "be able to run a query" in {
+    val query = s"""select * from ${BigQuerySchema.datasetName}.${BigQuerySchema.tableName} where amount = 9999 and event_timestamp > TIMESTAMP("2020-12-14 00:20:00");"""
+    val queryConfig = QueryJobConfiguration.newBuilder(query).build
+
+    val tableResult = bigQuery.query(queryConfig)
```

Comment:
do we need query functionality to insert something?  I can't see where this is used other than the test

Reply:
perhaps we could combine this with the below test, and query the record we just inserted?

Reply:
I added that test while developing just to test the connection was working correctly, I'm happy to remove it if you think it would be better?

Reply:
probably not useful in itself, although I did wonder if the "insert" call returns success, how do we know it actually arrived in bigquery, unless we query back the record.
So my suggestion would be remove this test, and query back in the "insert" test to be sure it actually arrived.